timemachine/parallel/client.py

```diff
@@ -2,21 +2,43 @@
 import multiprocessing
 import os
 import pickle
+from abc import ABC, abstractmethod
 from concurrent import futures
 from pathlib import Path
-from typing import List, Optional
+from typing import Any, Iterator, List, Optional, Sequence
+from uuid import uuid4
 
 from timemachine.parallel.utils import get_gpu_count
 
 # (ytz): The classes in this file are designed to help provide a consistent API between
 # multiprocessing (typically for local cluster use) and gRPC (distributed and multi-node).
 
 
+class BaseFuture(ABC):
+    @abstractmethod
+    def done(self) -> bool:
+        ...
+
+    @abstractmethod
+    def result(self) -> Any:
```

Comment:
nit: suggest using a `TypeVar` rather than returning `Any` here. This could help catch errors where we accidentally mix futures returning different types of results.

Reply:
It is actually desirable in my mind to be able to have a mix of futures. They might be doing completely different tasks

Reply:
To clarify, I just mean that this could prevent errors of the form:

```
def foo(fut: BaseFuture[Optional[int]]):
    return fut.result() + 1
```

With the type parameter, this sort of thing would fail static checks (because `1 + None` isn't allowed), but as-is it wouldn't.

(non-blocking)